web/pages/api/v0/bet/index.ts

```diff
@@ -1,24 +1,308 @@
-import { NextApiRequest, NextApiResponse } from 'next'
+import * as admin from 'firebase-admin'
+import { z } from 'zod'
 import {
-  CORS_ORIGIN_MANIFOLD,
-  CORS_ORIGIN_LOCALHOST,
-} from 'common/envs/constants'
-import { applyCorsHeaders } from 'web/lib/api/cors'
-import { fetchBackend, forwardResponse } from 'web/lib/api/proxy'
-
-export const config = { api: { bodyParser: false } }
-
-export default async function route(req: NextApiRequest, res: NextApiResponse) {
-  await applyCorsHeaders(req, res, {
-    origin: [CORS_ORIGIN_MANIFOLD, CORS_ORIGIN_LOCALHOST],
-    methods: 'POST',
-    allowedHeaders: 'Authorization',
+  DocumentReference,
+  FieldValue,
+  Query,
+  Transaction,
+} from 'firebase-admin/firestore'
+import { groupBy, mapValues, sumBy, uniq } from 'lodash'
+
+import { redeemShares } from 'functions/src/redeem-shares'
```

Comment:
Not sure what to do about this. Could copy the redemption code here.

Or, if we move it into `common`, then common has to depend on Firebase admin, which seems not ideal.

I could also add a TODO, because it won't be a problem if we decide to not proceed with Vercel functions. (Or, if we decide to migrate them all.)

Reply:
I would say it's totally unacceptable for `common` to depend on `firebase-admin`, because that would pull it into our website bundle. I think you should just factor out the "meat" of the redemption code as appropriate, or since this is just an experiment for now just copying it seems fine. That's much better than having this weird cross-dependency.